src/io/config_auto.cpp

```diff
@@ -294,6 +294,9 @@ const std::unordered_set<std::string>& Config::parameter_set() {
   "gpu_platform_id",
   "gpu_device_id",
   "gpu_use_dp",
+#ifdef USE_CUDA
+  "num_gpu", /* LGBM_CUDA */
+#endif
```

Comment:
I remember these codes are auto-generated, so we cannot manually change it.
@StrikerRUS to confirm.

Reply:
Right. Possibly should be in `config.cpp`